compose/runtime/runtime/build.gradle

```diff
@@ -116,9 +116,16 @@ if(AndroidXComposePlugin.isMultiplatformEnabled(project)) {
             commonMain.dependencies {
                 implementation(libs.kotlinStdlibCommon)
                 implementation(libs.kotlinCoroutinesCore)
+                implementation(project(":performance:performance-annotation"))
+
+                // Keep both project and module dependency for pseudo-relocated modules.
+                // The projects are actually empty, but we need them here just to
+                // make sure their maven coordinates are added to dependencies graph in user projects.
+                // It's a workaround until https://youtrack.jetbrains.com/issue/KT-61096/KLIB-Resolve-Drop-KLIB-resolve-inside-the-Kotlin-compiler
                 implementation(project(":annotation:annotation"))
                 implementation(project(":collection:collection"))
-                implementation(project(":performance:performance-annotation"))
+                implementation("androidx.annotation:annotation:1.9.1")
+                implementation("androidx.collection:collection:1.5.0-beta02")
```

Comment:
1. Use a version from catalog file
2. Can we use Google version for all modules?

Reply:
> Can we use Google version for all modules?

I think yes. We don't need to use a project dependency in all other modules. 
Compose runtime is unavoidable dependency of any Compose project.

Reply:
I mean I still don't fully understand why we should continue to publish it in case of removing all dependencies to it

Reply:
> I still don't fully understand why we should continue to publish it in case of removing all dependencies to it

There are many published 3rd party klibs which depend on compose-runtime 1.7.x and older. 

Those 3rd party klibs contain a list of all dependencies (including transitive dependencies). The list contains the unique names of the dependencies-klibs, including `org.jetbrains.compose.collection-internal` that we were publishing.

When Kotlin consumes one klib, it expects that all the dependencies are supplied (paths to those dependencies-klibs).
If we stop publishing our dumb klib with unique name `org.jetbrains.compose.collection-internal`, then no klib path would be provided and Kotlin would report an error about missing klib.

Reply:
Does this mean that we just need to publish only one no-op version and then remove it from publishing?

Reply:
> Those 3rd party klibs contain a list of all dependencies (including transitive dependencies). The list contains the unique names of the dependencies-klibs, including org.jetbrains.compose.collection-internal that we were publishing.

So, how are we going to drop experimental dependencies in the future? Like `ui-backhandler` that is supposed to be replaced by Google one once it will be there

Reply:
> Does this mean that we just need to publish only one no-op version and then remove it from publishing?

Not that simple unfortunately.
In order to be provided to Kotlin, the dependency must be present in the dependency graph (configured by gradle). 

Yes, we can stop publishing the dumb klib. But we will need to pin the dependency on the previously published dumb klib.

Reply:
> So, how are we going to drop experimental dependencies in the future? Like ui-backhandler that is supposed to be replaced by Google one once it will be there

We'll need the "dumb klib" approach too.  The published dumb klib version can be used as a pinned dependency, so it will be present in the dependencies graph.

Reply:
> It will be possible after https://youtrack.jetbrains.com/issue/KT-61096/KLIB-Resolve-Drop-KLIB-resolve-inside-the-Kotlin-compiler is solved.

As we discussed privately, this won't solve the issue. We should publish it until we decide to drop backward compatibility.

A case explaining the issue:
```
App -> Compose 1.8 -> android.compose.annotation 1.5
App -> SomeLib -> Compose 1.7 -> org.jetbrains.compose.annotation-internal 1.7
```
Without a dumb lib:
```
App -> Compose 1.8 -> org.jetbrains.compose.annotation-internal 1.8
```
It will always lead to duplicated classes

> Yes, we can stop publishing the dumb klib. 

We also discussed it. If we stop publishing klib, in Compose 1.9, for example, Compose 1.9 again won't be able to consume libs dependent on Compose 1.7

Reply:
What Ivan and I mean is that we can stop publishing the dumb klib and add a module dependency on the dumb klib, while also adding a module dependency on the latest androidx.annotation. So the dependency graph would look like this:

```
org.jetbrains.compose.runtime:1.9  // future version
     | -> org.jetbrains.compose.annotation-internal:1.8.0 // dumb klib. The version we'll publish with CM 1.8.0. 
                | -> androidx.annotation:1.9.1
     | -> androidx.annotation:2.0 // 2.0 is just an example of future version
```
This way androidx.annotation:2.0 would be chosen.

___

Right?

Reply:
> that we can stop publishing the dumb klib and add a module dependency on the dumb klib

Indeed, this works. Then we can stop publishing after the next stable release